src/utilstrencodings.cpp

```diff
@@ -538,3 +538,118 @@ int atoi(const std::string& str)
 {
     return atoi(str.c_str());
 }
+
+/** Upper bound for mantissa.
+ * 10^18-1, largest arbitrary decimal that will fit in 64 bit.
```

Comment:
it's worded somewhat ackwardly: my point is that larger 64-bit signed integers cannot consist of arbitrary combinations of 0-9:
```
 999999999999999999   1^18-1
9223372036854775807  (1<<63)-1 (max)
9999999999999999999  would not fit
```
Wouldn't be impossible to handle it, but I it would complicate bounds checking and don't see the point in our case.
I'll update the comment.